api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -0,0 +1,92 @@
+/*Minio Java Library for Amazon S3 Compatible Cloud Storage,(C)2015 Minio,Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+
+import io.minio.errors.InvalidArgumentException;
+
+/**
+ * A container class to hold all the Conditions to be checked before copying an object.
+ */
+public class CopyConditions {
+
+  private Map<String, String> copyConditions = new HashMap<>();
```

Comment:
If so, why am I not getting errors when  `./gradlew build`.  I use oracle jdk8 `1.8.0_111`.